server/service/src/programs/patient/insert_patient.rs

```diff
@@ -71,6 +79,13 @@ fn generate(input: InsertPatient, store_id: &str) -> NameRow {
         last_name,
         gender,
         date_of_birth,
+        address1,
+        phone,
+        date_of_death,
+        is_deceased: match is_deceased {
+            Some(is_deceased) => is_deceased,
+            None => false,
+        },
```

Comment:
I'm not sure if this is the best approach. This saves is_deceased as 'false' if no value is supplier to the server.
This is because the is_deceased value is not optional in the server but might not be supplied to the server.

I wonder if this might cause issues if for whatever reason the is_deceased value isn't given to the front end properly when update a patient, which would cause false 'alives' in the server.

I haven't quite wrapped my head around this and don't know if it is actually a risk.

Alternative option could be to make the is_deceased not optional, however currently isDeceased isn't part of the default patient json schema

Reply:
I wouldn't make it optional for the insert since the column is also required in the DB...

Similar for the update, we just upsert the whole row, i.e. the is_deceased flag must be passed in every time. Btw. this is because we use the `treat_none_as_null` option:

```rust
#[derive(Clone, Queryable, Insertable, Debug, PartialEq, Eq, AsChangeset, Default)]
#[changeset_options(treat_none_as_null = "true")]
#[table_name = "name"]
pub struct NameRow {
```

Another note, if we wouldn't use `treat_none_as_null` we couldn't easily distinguish between leaving a field unset, updating the value or setting the column to null. To carry all this information we would need something like `is_deceased: Option<Option<bool>>` where:
- `is_desceased: None` would mean to leave the column unchanged
- `is_desceased`: Some(None)` would mean to unset the column to null
- `is_desceased`: Some(true|false)` would mean to set the column to true or false

Reply:
Please have a look `service/src/stocktake/insert.rs`. Please ignore parts of my previous comment. For consistency please just copy how it is done there.